toolchain/parse/node_kind.h

```diff
@@ -119,20 +145,69 @@ class NodeKind::Definition : public NodeKind {
   Definition(const Definition&) = delete;
   auto operator=(const Definition&) -> Definition& = delete;
 
+  // Returns true if the node is bracketed; otherwise, child_count is used.
+  constexpr auto has_bracket() const -> bool { return bracket_ != *this; }
```

Comment:
It looks like you're changing the semantic so that both a bracket and child_count can be provided. I can see why you're doing that, but I think this now makes the documentation unclear: "otherwise" implies to me mutually exclusive.

Please clarify comments about the relationship between the attributes. It might be worth documenting at the class level, given the scope of the relationship.

Reply:
Done.